staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -304,8 +344,27 @@ func interfaceFromUnstructured(sv, dv reflect.Value) error {
 	return nil
 }
 
-func (c *Converter) ToUnstructured(obj runtime.Object, u *map[string]interface{}) error {
-	return toUnstructured(reflect.ValueOf(obj).Elem(), reflect.ValueOf(u).Elem())
+func (c *ConverterImpl) ToUnstructured(obj runtime.Object, u *map[string]interface{}) error {
+	err := toUnstructured(reflect.ValueOf(obj).Elem(), reflect.ValueOf(u).Elem())
+	if c.mismatchDetection {
+		newUnstr := &map[string]interface{}{}
+		newErr := toUnstructuredViaJSON(obj, newUnstr)
+		if (err != nil) != (newErr != nil) {
+			glog.Fatalf("ToUnstructured unexpected error: %v, expected: %v", err, newErr)
```

Comment:
Fatal doesn't give the call stack - it calls os.Exit().

Error messages (as we have them) are fairly bad at providing the exact context for the error.  If we have this on and are debugging it in a large scale test, I want all the possible info contained in the output.